### ISSUE TITLE
Which relationships are used for phenotype expressions

### ISSUE THREAD

**Issue by hlapp** [phenoscape/phenoscape-kb-services#16]
The OntoTrace API documentation gives as example for then entity query parameter an expression of the form `part_of some E`. There doesn't seem to be documentation on properties other than `part_of` that one might want to use for querying (`develops_from` for example?), and how one would best combine these if combining them is even possible through a superproperty.

cc @xu-hong

**Comment by balhoff** [comment]
Keep in mind you are selecting anatomical entities, not phenotypes. The short answer is that you can provide any expression you want. The way to know what is useful is to know about the contents of Uberon and what the Elk reasoner will do for you when applying that expression to Uberon. Is that the sort of context I should add to the API docs? Or some more specific examples? Combinations are possible by chaining in the expression.

**Comment by hlapp** [comment]
> Keep in mind you are selecting anatomical entities, not phenotypes.

Right, I fixed the title.

> The short answer is that you can provide any expression you want. The way to know what is useful is to know about the contents of Uberon and what the Elk reasoner will do for you when applying that expression to Uberon.

The way to know what is useful is not to know what is in Uberon but to know what is in the database. There are dozens of relationships in Uberon (or RO) but only for less than a handful of them will OntoTrace return any data. Properties for which there are no data aren't useful choices to offer to a user. Which are the few that have data? And if I wanted something like "any property restriction" is there a superproperty that accomplishes that (`owl:topObjectProperty`? don't you also compute some `involved_in some X` superclass?).

**Comment by balhoff** [comment]
I agree that the documentation should provide some guidance, but the relationship to the data is not as tight as you say. Of course it does matter which entities you end up selecting, but you could select them by any property used in Uberon. You could ask for anything that `develops_from some cartilage`, although `develops_from` is not really used within Phenoscape data annotations. If you used `topObjectProperty`, you would end up with every possible anatomical structure. Maybe the documentation should make clear that this essentially is doing a query to Uberon that doesn't really interact with the data annotations at all (with the exception of bringing in some postcompositions). The results you get are limited to which of the resulting entities are related to phenotypes in the KB, but there isn't an a priori way to say which properties have a relation to the data.

**Comment by hlapp** [comment]
> Maybe the documentation should make clear that this essentially is doing a query to Uberon that doesn't really interact with the data annotations at all

Yes indeed that would be good. I see now where my mistake was. I think it would generally help to document expressly how input OWL expressions are used to select matches.

> The results you get are limited to which of the resulting entities are related to phenotypes in the KB, but there isn't an a priori way to say which properties have a relation to the data.

Right, that would depend on the entity that's the object of the property restriction.

That said, short of hard-coding the relationships used in Uberon, is there a good query method to obtain the properties used in a given ontology, here Uberon?

**Comment by balhoff** [comment]
There aren't any queries for properties in the Phenoscape API. There is a property listing at Ontobee: http://www.ontobee.org/browser/term.php?o=UBERON&iri=http://www.w3.org/2002/07/owl%23ObjectProperty&graph=http://purl.obolibrary.org/obo/merged/UBERON

You can click on a property and see the complete list of axioms using the property (but for some reason it doesn't seem to be working correctly for `part_of`).

**Comment by hlapp** [comment]
> There aren't any queries for properties in the Phenoscape API.

Can we add one then? Do you want me to post this as a separate issue?

**Comment by balhoff** [comment]
Sure, a separate issue would be good. It would be good to know exactly what you have in mind, because some of the queries I can imagine would be kind of a challenge.
